ethereumj-core/src/main/java/org/ethereum/core/Transaction.java

```diff
@@ -252,10 +255,9 @@ public boolean isParsed() {
 
     public byte[] getHash() {
         if (!isEmpty(hash)) return hash;
-
         rlpParse();
-        byte[] plainMsg = this.getEncoded();
-        return HashUtil.sha3(plainMsg);
+        getEncoded();
+        return hash;
```

Comment:
This change causes an empty array to be returned when parsed == true && rlpEncoded != null. Happends all the time with the StandaloneBlockchain.

Please revert this method to:

`public byte[] getHash() {
        if (!isEmpty(hash)) return hash;
         rlpParse();
        byte[] plainMsg = this.getEncoded();
        return HashUtil.sha3(plainMsg);
}`